addons/web_editor/static/src/js/common/utils.js

```diff
@@ -185,13 +185,28 @@ function _computeColorClasses(colorNames, prefix = 'bg-') {
  * @returns {string}
  */
 function _getCSSVariableValue(key, htmlStyle) {
+    let iframeValue = undefined;
     if (htmlStyle === undefined) {
         htmlStyle = window.getComputedStyle(document.documentElement);
+        // There could be a better way of doing it.
+        if (window.length > 1) {
+            for (let i = 0; i < window.length; i++) {
+                const iframeWindow = window[i];
+                if (iframeWindow.frameElement && iframeWindow.frameElement.classList.contains('wysiwyg_iframe')) {
+                    const doc = iframeWindow.document;
+                    const tempEl = doc.body.appendChild(doc.createElement('div'));
+                    tempEl.className = `bg-${key}`;
+                    iframeValue = iframeWindow.getComputedStyle(tempEl).getPropertyValue("background-color").trim();
+                    tempEl.parentNode.removeChild(tempEl);
```

Comment:
Can't we do this properly by adding the required CSS variable on the iframe HTML when needed and reading values from those like in the website case?

Reply:
I checked this possibility but the situation here is that some default theme color classes (e.g. '.bg-o-color-4') are overridden by CSS using a template related class (e.g. '.o_resto_theme') to set new values on them, so I don't really see how we can do it in a clean way :thinking:

Reply:
Those damn mailing templates :/ But so this is not really related to the fact that this is inside an iframe so this patch feels really bad :grimacing: What about overriding the method which calls this `_getCSSVariableValue` when inside the mailing editor instead? That would be the proper way to do it I think.

Reply:
Actually the `_getCSSVariableValue` is used in option's methods, so I updated the code to override its behaviour inside mass mailing iframe.

Reply:
I understand that, my point was to override the option's behavior to not use `_getCSSVariableValue` (since you don't want a CSS variable there)